app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,13 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(params.require(:comment).permit(:text))
+    @comment.author = current_user
+    if @comment.save
+      redirect_to user_post_url(@post.author, @post), notice: 'Comment created successfully.'
+    else
+      flash.now[:error] = 'Error: Comment could not be saved'
+      redirect_to user_post_url(@post.author, @post)
+    end
+  end
```

Comment:
- [ ] [OPTIONAL] As a suggestion I would like to recommend you add `the params method` , you can read more in this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)  👍 

```ruby
  private 
  def comment_params
    params.require(:comment).permit(:text)
  end
```